source/core/errors.ts

```diff
@@ -89,8 +89,8 @@ An error to be thrown when the server response code is not 2xx nor 3xx if `optio
 Includes a `response` property.
 */
 // eslint-disable-next-line @typescript-eslint/naming-convention
-export class HTTPError extends RequestError {
-	declare readonly response: Response;
+export class HTTPError<T = any> extends RequestError<T> {
```

Comment:
```suggestion
export class HTTPError<T = unknown> extends RequestError<T> {
```

Reply:
This will be a breaking change, isn't it?
eg this code will no more work
```ts
await got('https://google.com', { responseType: 'json' })
  .catch((err: HTTPError) => console.log(err.response.body.foo)) // 'err.response.body' is of type 'unknown'.ts(18046)

```

Reply:
Right. Add a code comment at least, like

> TODO: Change this to `unknown` in the next major version.